server/channels/api4/outgoing_oauth_connection.go

```diff
@@ -92,6 +116,10 @@ func listOutgoingOAuthConnections(c *Context, w http.ResponseWriter, r *http.Req
 		return
 	}
 
+	if !checkOutgoingOAuthConnectionReadPermissions(c) {
```

Comment:
Any reason for not having the permission check as first statement? I wouldn't expect to check anything else if the user is not allowed to go ahead with the request.

Reply:
not really, in some cases it was more to check the input parameter before I used it in audits, but it can work either way. Let me move things a bit here.